src/core/environment.go

```diff
@@ -868,12 +868,43 @@ func virtualization() (string, string) {
 		return "", "host"
 	}
 
-	if virtualizationSystem == "docker" {
+	if virtualizationSystem == "docker" || isContainer() {
+		log.Debugf("Virtualization detected as container with role %v", virtualizationRole)
 		return "container", virtualizationRole
 	}
+	log.Debugf("Virtualization detected as %v with role %v", virtualizationSystem, virtualizationRole)
 	return virtualizationSystem, virtualizationRole
 }
 
+func isContainer() bool {
```

Comment:
```suggestion
func (env *EnvironmentType) virtualization() (string, string) {
	...

	if virtualizationSystem == "docker" || env.IsContainer() {
   ...
}
```

Reply:
suggest amending the virtualization function instead and reusing env.IsContainer instead